AppInspector.CLI/CLICmdOptions.cs

```diff
@@ -2,46 +2,42 @@
 // Licensed under the MIT License. See LICENSE.txt in the project root for license information.
 
 using CommandLine;
-using NLog;
+using Microsoft.ApplicationInspector.Commands;
 
-namespace Microsoft.ApplicationInspector.Commands
+namespace Microsoft.ApplicationInspector.CLI
 {
-    public class AllCommandOptions
+    /// <summary>
+    /// CLI command option classes add output arguments to common properties for each command verb
+    /// </summary>
+    /// 
+    public class CLICommandOptions : CommandOptions
     {
-        public Logger Log { get; set; }
+        [Option('o', "output-file-path", Required = false, HelpText = "Output file path")]
+        public string OutputFilePath { get; set; }
+        [Option('f', "output-file-format", Required = false, HelpText = "Output format [json|text]", Default = "text")]
+        public string OutputFileFormat { get; set; }
 
-        [Option('l', "log-file-path", Required = false, HelpText = "Log file path")]
-        public string LogFilePath { get; set; }
-        [Option('v', "log-file-level", Required = false, HelpText = "Log file level [Debug|Info|Warn|Error|Trace|Fatal|Off]", Default = "Error")]
-        public string LogFileLevel { get; set; }
+        public CLICommandOptions()
+        {
+            OutputFileFormat = "text";
+        }
+    }
 
-        public bool CloseLogOnCommandExit { get; set; }
 
-    }
     /// <summary>
-    /// Command option classes for each command verb
+    /// CLI command distinct arguments 
     /// </summary>
-
     [Verb("analyze", HelpText = "Inspect source directory/file/compressed file (.tgz|zip) against defined characteristics")]
-    public class AnalyzeCommandOptions : AllCommandOptions
+    public class CLIAnalyzeCmdOptions : CLICommandOptions
     {
         [Option('s', "source-path", Required = true, HelpText = "Source file or directory to inspect")]
         public string SourcePath { get; set; }
 
-        [Option('o', "output-file-path", Required = false, HelpText = "Output file path")]
-        public string OutputFilePath { get; set; }
-
-        [Option('f', "output-file-format", Required = false, HelpText = "Output format [html|json|text]", Default = "html")]
-        public string OutputFileFormat { get; set; }
-
-        [Option('e', "text-format", Required = false, HelpText = "Match text format specifiers", Default = "Tag:%T,Rule:%N,Ruleid:%R,Confidence:%X,File:%F,Sourcetype:%t,Line:%L,Sample:%m")]
-        public string TextOutputFormat { get; set; }
-
         [Option('r', "custom-rules-path", Required = false, HelpText = "Custom rules file or directory path")]
         public string CustomRulesPath { get; set; }
 
-        [Option('t', "tag-output-only", Required = false, HelpText = "Output only identified tags", Default = false)]
-        public bool SimpleTagsOnly { get; set; }
+        [Option('h', "match-depth", Required = false, HelpText = "First match or best match based on confidence level (first|best)", Default = "best")]
```

Comment:
This should be `string?`.

Only marking one of these.  https://docs.microsoft.com/en-us/dotnet/csharp/tutorials/nullable-reference-types

Reply:
If there were more than two possible argument values then yes but in this case all that is needed is a binary yes/no and a boolean is correct.

Reply:
Sorry, see other comment. This actually refers to the line below, which is a `string` but you're not setting a default value so it is actually a `string?`.  Since this is a big rewrite it seems like a good opportunity to do nullable annotations.

Reply:
Ah -fixed.  Yes on nullables but separating refactoring first as per priorities discussion but it's next

Reply:
For CLI the defaults specified in the properties do the work but in cases where the property is overridden from the base the CommandLineParser doesn't set them right so I do in the derived class constructors.  This is fixed for all missing in this class.